src/identity.js

```diff
@@ -611,6 +615,76 @@ export class Identity extends EventEmitter {
         throw new SDKError('The user is not connected to this merchant');
     }
 
+    /**
+     * @async
+     * @function
+     * @summary
+     * Retrieves the external identifier (`externalId`) for the authenticated user.
+     *
+     * In Schibsted Account there are multiple ways of identifying users, however for integrations with
+     * third-parties it's recommended to use `externalId` as it does not disclose
+     * any critical data whilst allowing for user identification.
+     *
+     * `externalId` is merchant-scoped using a pairwise identifier (`pairId`),
+     * meaning the same user's ID will differ between merchants.
+     * Additionally, this identifier is bound to the external party provided as argument.
+     *
+     * @description This function calls {@link Identity#hasSession} internally and thus has the side
+     * effect that it might perform an auto-login on the user
+     * @throws {SDKError} If the `pairId` is missing in user session.
+     * @throws {SDKError} If the `externalParty` is not defined
+     * @return {Promise<string>} The merchant- and 3rd-party-specific `externalId`
+     */
+    async getExternalId(externalParty, optionalSuffix = "") {
+        if(!externalParty || externalParty.length === 0) {
+            throw new SDKError('externalParty cannot be empty');
+        }
+        const _toHexDigest = (hashBuffer) =>{
+            // convert buffer to byte array
+            const hashArray = Array.from(new Uint8Array(hashBuffer));
+            // convert bytes to hex string
+            return hashArray.map(b => b.toString(16).padStart(2, '0')).join('');
+        }
+
+        const _getSha256Digest = (data) => {
+            return crypto.subtle.digest('SHA-256', data);
+        }
+
+        const _hashMessage = async (message) => {
+            const msgUint8 = new TextEncoder().encode(message);
+            return _getSha256Digest(msgUint8).then( (it) => _toHexDigest(it));
+        }
+
+        const _constructMessage = (pairId, externalParty, optionalSuffix) => {
+            return optionalSuffix
+                ? `${pairId}:${externalParty}:${optionalSuffix}`
+                : `${pairId}:${externalParty}`;
+        }
+
+        const { pairId } = await this.hasSession();
+
+        if (!pairId)
+            throw new SDKError('pairId missing in user session!');
```

Comment:
I think this can be moved at the beginning, near the externalParty check, to avoid preparing all the cryptography if user doesn't have a session